src/components/PlanGate.tsx

```diff
@@ -0,0 +1,53 @@
+'use client'
+
+import { ReactNode, useEffect, useState } from 'react'
+import { useKeyring, Plan } from '@w3ui/react-keyring';
+import StripePricingTable from './PricingTable';
+import DefaultLoader from './Loader';
+
+export function PlanGate ({ children }: { children: ReactNode }): ReactNode {
+  const [error, setError] = useState<any>()
+  const [{ account }, { getPlan }] = useKeyring();
+  const [plan, setPlan] = useState<Plan>()
+  useEffect(function () {
+    (async function () {
+      if (account) {
+        try {
+          const result = await getPlan(account as `${string}@${string}`)
+          if (result.ok) {
+            setPlan(result.ok)
+          } else {
+            setError(result.error)
+          }
+        } catch (err) {
+          console.error("CAUGHT ERROR", err)
+          setError(err)
+        }
+      }
+    })()
+  }, [account, getPlan])
+  if (!plan && !error) {
+    return <DefaultLoader className='w-12 h-12 text-white' />
+  }
+
+  if ((error?.name === 'PlanNotFound') || (plan && !plan.product)){
```

Comment:
```suggestion
  if (!plan?.product) {
```

Reply:
hm why's that? I think it's totally possible to get a `PlanNotFound` error here and I think we want to send people through the plan picking process if so...

Reply:
Just thought it was simpler - I assume if the plan was not found there will also not be a `product` property on this `plan` object...

Reply:
oh that's probably right yea, will double check that in the code but I think you're dead-on that `plan` being null-ish also covers the case where `error.name === 'PlanNotFound'`, thx!